### PR TITLE
Multi tenant RavenDB sample

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

File: samples/multi-tenant/ravendb/sample.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+include: persistence-code
+
+
+#### Outbox cleanup
+
+The built-in Outbox cleanup does not work in a multi-tenant environment because it executes in the context of the shared database while the Outbox documents are stored in the tenant databases. For that reason it needs to be disabled.
+
+snippet: DisableOutboxCleanup
+
+Alternative solution needs to be configured. The simplest way to ensure that the dispatched Outbox documents are removed is to use the RavenDB [Document expiration](https://ravendb.net/docs/article-page/3.5/Csharp/server/bundles/expiration) bundle. Following code ensures that tenant databases are created with this bundle enabled:
```
Comment (mauroservienti) [inline samples/multi-tenant/ravendb/sample.md]:
```suggestion
The simplest way to ensure that the dispatched Outbox documents are removed is to use the RavenDB [Document expiration](https://ravendb.net/docs/article-page/3.5/Csharp/server/bundles/expiration) bundle. Following code ensures that tenants’ databases are created with this bundle enabled:
```

File: samples/multi-tenant/ravendb/Core_7/Receiver/MarkForExpiryListener.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class MarkForExpiryListener : IDocumentStoreListener
+{
+    public bool BeforeStore(string key, object entityInstance, RavenJObject metadata, RavenJObject original)
+    {
+        if (entityInstance.GetType().Name != "OutboxRecord")
+        {
+            return false;
+        }
+
+        var dispatched = entityInstance.GetPropertyValue<bool>("Dispatched");
```
[the remark this answers is not in the document]
Comment (SzymonPobiega) [reply samples/multi-tenant/ravendb/Core_7/Receiver/MarkForExpiryListener.cs]: Not really. The `OutboxRecord` type is internal :-(

File: samples/multi-tenant/ravendb/Core_7/Receiver/Order.cs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+﻿public class Order
+{
+    public virtual string Id { get; set; }
```
[the remark this answers is not in the document]
Comment (SzymonPobiega) [reply samples/multi-tenant/ravendb/Core_7/Receiver/Order.cs]: 👍

**Comment by SzymonPobiega** [comment]
@mauroservienti code and doco fixed.

**Pull request merged**